core/src/test/java/org/elasticsearch/action/update/UpdateRequestTests.java

```diff
@@ -322,6 +353,38 @@ public void testNowInScript() throws IOException {
         }
     }
 
+    public void testTimeout() {
+        final GetResult getResult =
+                new GetResult("test", "type", "1", 0, true, new BytesArray("{\"f\":\"v\"}"), null);
+
+        final boolean delete = randomBoolean();
+        final Script script =
+                delete ? mockInlineScript("ctx.op = delete") : mockInlineScript("return");
+        final UpdateRequest updateRequest =
+                new UpdateRequest("test", "type", "1")
+                        .script(script)
+                        .timeout(randomTimeValue());
+        final UpdateHelper.Result result = updateHelper.prepare(
+                new ShardId("test", "", 0),
+                updateRequest,
+                getResult,
+                ESTestCase::randomNonNegativeLong);
+        final Streamable action = result.action();
+        if (delete) {
+            assertThat(action, instanceOf(DeleteRequest.class));
+            final DeleteRequest deleteRequest = (DeleteRequest) action;
+            assertThat(deleteRequest.timeout(), equalTo(deleteRequest.timeout()));
+        } else {
+            assertThat(action, instanceOf(IndexRequest.class));
+            final IndexRequest indexRequest = (IndexRequest) action;
+            assertThat(indexRequest.timeout(), equalTo(updateRequest.timeout()));
+        }
+    }
+
+    private Script mockInlineScript(final String script) {
```

Comment:
If we think this makes things more readable we should probably add it as a static method on `MockScriptEngine` and statically import it in the tests.